### ISSUE TITLE
modal screen bugs bars and vbars

### ISSUE THREAD

**Issue by Elckarow** [renpy/renpy#3988]
say you have a very simple gallery screen
```
screen gallery():
    viewport id "gallery_vp":
        draggable True
        xalign 0.0
        vbox:
            for cg in cg_list:
                imagebutton:
                    idle cg.small
                    action Show("cg_display", cg=cg.cg)
    
    vbar value YScrollValue("gallery_vp")
```
and that `cg_display` is a modal screen the vbar becomes unresponsive upon hiding the screen. It can still be dragged, but it won't chage the viewport's adjustment, and dragging the viewport won't make the bar adjust itself.

**Comment by Elckarow** [comment]
after some more testing, the bar _does_ become responsive once you click anywhere on the gallery screen after hiding the modal screen

show gallery screen
show modal screen
hide modal screen
bar is unresponsive
click somewhere
bar is responsive

**Comment by renpytom** [comment]
I'm not able to reproduce this issue. Here's my replication, based on yours.

```renpy
define _confirm_quit = False

define e = Character("Eileen", image="eileen")
define l = Character("Lucy", image="lucy")
define a = Character("Augustina", image="augustina")

label main_menu:
    return


screen gallery():
    default content = list(range(100))

    viewport id "gallery_vp":
        draggable True
        xalign 0.0
        vbox:
            for i in content:
                textbutton "[i]":
                    xpos 100
                    action Show("cg_display")

    vbar value YScrollValue("gallery_vp")

screen cg_display():
    modal True

    textbutton "Hide":
        xalign 1.0
        action Hide("cg_display")

label start:

    scene bg washington
    call screen gallery

    "..."

    return
```

**Comment by Elckarow** [comment]
here's a [build](https://www.mediafire.com/file/qooafwg39pnf2h4/BugModal-1.0-pc.zip/file) where the issue is present (the gallery screen)

the code is line 288, right above the navigation screen

this is in renpy 8.0.3, but i fist caught wind of this bug in 7.4.5 with this exact same gallery code

**Comment by renpytom** [comment]
I just landed a fix to this. It was a bit confusing since the modal screen wasn't the problem - it was the dissolve when the screen closed, but this should be fixed in the next nightly.

**Comment by mal** [comment]
Aforementioned fix 9ac23face8449a9cd6558828def49f9c41f60649.
